### DISCUSSION TITLE
How to convert string to lowercase in move?

### DISCUSSION THREAD

**Discussion by 0xmaayan** [aptos-labs/aptos-developer-discussions#205] [Questions] [answered]
### Discord user ID

0xN0rd

### Describe your question in detail.

hi, how to convert string to lowercase in move?
`.to_lowercase()` is not working

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by CapCap** [comment, marked as answer]
Hi! There isn't a built in yet, but something like the following would work for ascii strings:

```
    use std::string::{Self, String};
    use std::vector;

    public fun to_lowercase(s: &String): String {
        let bytes = string::bytes(s);
        let lowercased_bytes = vector::map_ref<u8, u8>(bytes, |byte| {
            let byte = *byte;
            if (byte >= 65 && byte <= 90) {
                byte + 32
            } else {
                byte
            }
        });
        string::utf8(lowercased_bytes)
    }
```
